readthedocs/search/parsers.py

```diff
@@ -140,21 +140,70 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
+                    yield {
+                        "id": _id,
+                        "title": title,
+                        "content": content,
+                    }
+                except Exception:
+                    log.info("Unable to index dt section.", exc_info=True)
+
+            # Remove the <dl> node
+            # There isn't a clear indication if this behavior is DFS or BFS
+            dl.decompose()
+
+        # This is pattern is intended for a future generic scenario
+        # We could include the <dl> pattern in _is_section etc. instead of having the above
+        # <dl> block
+        outer_sections = []
         # Index content from h1 to h6 headers.
         for head_level in range(1, 7):
-            tags = body.css(f'h{head_level}')
-            for tag in tags:
+            outer_sections.append(body.css(f"h{head_level}"))
```

Comment:
Looks like we are looping twice for no reason?

Reply:
See the new comment above. This is just to accomodate the pattern. I think we can be more generic. So I left it open for additions. I was thinking that the addition of the `<dl>` block is waiting for the next chance to be improved and generalized and merged together with this block.

Reply:
Turned it into a 1-liner :)

Reply:
Doesn't look like that we need it now, and it's iterating the nodes twice. We should add this code only it's needed now.